source/game_sa/CTimer.h

```diff
@@ -7,40 +7,50 @@ Do not delete this comment block. Respect others' work!
 #pragma once
 
 #include "PluginBase.h"
+#include <array>
 
 class PLUGIN_API CTimer
 {
 public:
+    using UpdateTimeHistory_t = std::array<unsigned int, 4>;
+
+    typedef std::uint64_t(__cdecl* TimerFunction_t)();
+    static TimerFunction_t& ms_fnTimerFunction;
+
     // class variables
-    static bool& bSkipProcessThisFrame;
-    static bool& bSlowMotionActive;
-    static float& game_FPS;
+    static bool& ms_bEnableTimeDebug;
+    static bool& ms_bSkipProcessThisFrame;
+    static bool& ms_bSlowMotionActive;
+    static float& ms_fGameFPS;
 
     static bool& m_CodePause;
-    static unsigned int& m_FrameCounter;
     static bool& m_UserPause;
+    static unsigned int& m_FrameCounter;
+    static unsigned int& ms_nRenderTimerPauseCount;
     static float& ms_fTimeStepNonClipped;
     static float& ms_fTimeStep;
+    static unsigned int& ms_nTimerDivider;
 
-    static unsigned int& m_snPPPPreviousTimeInMilliseconds;
-    static unsigned int& m_snPPPreviousTimeInMilliseconds;
-    static unsigned int& m_snPPreviousTimeInMilliseconds;
-    static unsigned int& m_snPreviousTimeInMillisecondsNonClipped;
     static float& ms_fOldTimeStep;
     static float& ms_fSlowMotionScale;
 
     // game speed
     static float& ms_fTimeScale;
-    static unsigned int& m_snPreviousTimeInMilliseconds;
     static unsigned int& m_snTimeInMillisecondsPauseMode;
     static unsigned int& m_snTimeInMillisecondsNonClipped;
+    static unsigned int& ms_nPreviousTimeInMillisecondsNonClipped;
     static unsigned int& m_snTimeInMilliseconds;
+    static std::uint64_t& m_snRenderStartTime;
 
-    // class functions
+    // Array of update time history with the last item being from before the most recent frame
+    static UpdateTimeHistory_t& m_UpdateTimeMsHistory;
+
+    static void InjectHooks();
 
+    // class functions
     static void Initialise();
     static void Shutdown();
-    static void UpdateVariables(float timeStep);
+    static void UpdateVariables(float timeStep); // Also called GetRealTimeScale in some idbs
```

Comment:
```suggestion
    static void UpdateVariables(float timeStep);
```
We only care about what it's named in the android version.

Reply:
It was named `GetRealTimeScale` in my IDB. 
Thats the one I've downloaded from your Discord.
I presume thats not the Android IDB then.

Reply:
You can download the android idb here: https://drive.google.com/open?id=1-fnNWSiT7pXGizv9rwNoX6BycO43j9HN